src/gallium/drivers/lima/lima_draw.c

```diff
@@ -380,11 +382,24 @@ lima_pack_plbu_cmd(struct lima_context *ctx, const struct pipe_draw_info *info)
       plbu_cmd[i++] = gl_position_va;
       plbu_cmd[i++] = 0x10000100; /* INDEXED_DEST */
 
-      struct lima_resource *res = lima_resource(info->index.resource);
+      struct pipe_resource *indexbuf = NULL;
+      unsigned index_offset = 0;
+      struct lima_resource *res;
+      if (info->has_user_indices) {
+         util_upload_index_buffer(&ctx->base, info, &indexbuf, &index_offset);
+         res = lima_resource(indexbuf);
+      }
+      else
+         res = lima_resource(info->index.resource);
+
       lima_bo_update(res->bo, false, true);
+      lima_submit_add_bo(ctx->gp_submit, res->bo, LIMA_SUBMIT_BO_READ);
       plbu_cmd[i++] = res->bo->va + info->start +
```

Comment:
Seems this address calculation is not right from util_upload_index_buffer implementation. And we may just use u_upload_data directly to remove the min_out_offset constraint to save space, although the address calculation will be different for has_user_indices true & false cases then.